.github/workflows/ci.yml

```diff
@@ -4,6 +4,10 @@ on:
   # Run workflow on fork repository will help contributors find and resolve issues before sending a PR.
   push:
   pull_request:
+# https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

Comment:
What's the value of `github.head_ref` exactly? I didn't get it from the [github-context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context):
> The head_ref or source branch of the pull request in a workflow run.

And I can't get an example from the documents.

Reply:
https://github.com/karmada-io/karmada/pull/3079#discussion_r1089875933

> https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value
> The github.head_ref is only defined on pull_request events. if github.head_ref is undefined, the concurrency group will fallback to the github.run_id, which is guaranteed to be both unique and defined for the run.
> Each PR has a unique github.head_ref, which ensures that it will be cancelled correctly, and for non-PR triggers ensures that they will not be cancelled.

Reply:
Can you give the example of the group name for both `push` and `pull_request` events?
We should ensure that each `pull request` uses a unique group name so that it won't cancel the workflows that are triggered by `push` event.

Reply:
It is in the [github-context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context).

```
github.head_ref | string | The head_ref or source branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is either pull_request or pull_request_target.
```

Reply:
echo ${{ github.workflow }}-${{ github.head_ref }}-${{ github.run_id }}

```
CI Workflow-ci/cancel-4071108226
```

Reply:
github.workflow: is the `.name` of the `ci.yml`
github.head_ref: is the branch of the pr
github.run_id: is the job run id

Reply:
I added `github.actor` to avoid wrong cancellation due to different users but same branch name

Reply:
> https://github.com/karmada-io/karmada/actions/runs/4071108226/jobs/7012597471
> 
> `echo ${{ github.workflow }}-${{ github.head_ref }}-${{ github.run_id }}`
> 
> ```
> CI Workflow-ci/cancel-4071108226
> ```

Do you mean we can get the group name from logs?

Reply:
Yeash